### ISSUE TITLE
Allow user to propose new time

### ISSUE THREAD

**Issue by dlpartain** [Ericsson/exchangecalendar#89]
Hi,

Though we are not necessarily trying to re-invent a competing technology, many users are used to being able to respond to a meeting invitation by proposing a new time.  It would be nice to be able to do this using this tool!

Thanks. 

David

**Comment by buzzz321** [comment]
Yeah it would be really nice since meeting juggling is the norm at workplaces

**Comment by bavincen** [comment]
the organizer of the event is able to see availability of attendees when typing mail address

**Comment by apetzo** [comment]
hi

that's not the point. There can be many reasons to propose a different time
for a meeting, even if my calendar is empty at the originally proposed
time. Having the ability to directly propose an alternative time would
really be useful.

Cheers,

Andreas

**Comment by muthusuba** [comment]
I would like this as well!

@apetzo A quick question: Would you know if Outlook or Exchange Server supports this, please?

**Comment by apetzo** [comment]
I only know that in Outlook you get the option to reject/tentatively accept
and propose a new time.

Andreas

**Comment by bavincen** [comment]
supported by 2013 server ..

**Comment by wkmanire** [comment]
I have a similar symptom which i believe is related to this same issue. When others send "new time proposal" meeting events to me I cannot accept them. Instead I see a message in Thunderbird warning me that "This message contains an event that this version of Lightning cannot process"

![screenshot from 2016-07-13 13-43-57](https://cloud.githubusercontent.com/assets/217010/16819150/ec8dc044-48ff-11e6-978c-f4f57522304a.png)

**Comment by wkmanire** [comment]
Is anyone able to take a look at this? This is an important feature for business users.

**Comment by bavincen** [comment]
this shows if you have read only calendar or the event delted from lightning

**Comment by bavincen** [comment]
let user only propose time for attendee
https://github.com/Ericsson/exchangecalendar/commit/e9d219b0e09374d37ee597fae643ff6ab1ca4d9c


organizer part has to be done
https://msdn.microsoft.com/en-us/library/office/dn495613(v=exchg.150).aspx

**Comment by bavincen** [comment]
The organizer will receive a MeetingResponse message when the attendee responds with a proposed new meeting time. The MeetingResponse message contains the proposed new meeting start time and end time, and the identifier of the associated calendar item in the organizer’s calendar. The organizer can use that information to update their existing calendar item for the meeting. The following is the workflow for the organizer to respond to a MeetingResponse message that proposes a new meeting time:

    Determine whether the ProposedStart or ProposedEnd elements have been set in the MeetingResponse. If so, go to step 2. If not, the MeetingResponse message only indicates whether the attendee has accepted, tentatively accepted, or declined the meeting.

    Get the organizer’s existing calendar item for the meeting by using the EWS identifier returned in the AssociatedCalendarItemId element.

    Compare the original start and end time with the proposed new meeting time. If the proposed new meeting time is acceptable to the organizer, go to step 4. Otherwise, the meeting organizer can either ignore the proposed meeting time, or send an email response to the attendee that proposed the new meeting time.

    (Optional) Perform a GetUserAvailability EWS operation call to find out whether the proposed time will work for all attendees, including room and resource mailboxes. (You can also use the ExchangeService.GetUserAvailability EWS Managed API method to do this.)

    The organizer can then update their meeting with the new proposed meeting times and send the updates to all attendees by using the UpdateItem EWS operation (or the Appointment.Update EWS Managed API method).

**Comment by bavincen** [comment]
The organizer will receive a MeetingResponse message when the attendee responds with a proposed new meeting time. The MeetingResponse message contains the proposed new meeting start time and end time, and the identifier of the associated calendar item in the organizer’s calendar. The organizer can use that information to update their existing calendar item for the meeting. The following is the workflow for the organizer to respond to a MeetingResponse message that proposes a new meeting time:

    Determine whether the ProposedStart or ProposedEnd elements have been set in the MeetingResponse. If so, go to step 2. If not, the MeetingResponse message only indicates whether the attendee has accepted, tentatively accepted, or declined the meeting.

    Get the organizer’s existing calendar item for the meeting by using the EWS identifier returned in the AssociatedCalendarItemId element.

    Compare the original start and end time with the proposed new meeting time. If the proposed new meeting time is acceptable to the organizer, go to step 4. Otherwise, the meeting organizer can either ignore the proposed meeting time, or send an email response to the attendee that proposed the new meeting time.

    (Optional) Perform a GetUserAvailability EWS operation call to find out whether the proposed time will work for all attendees, including room and resource mailboxes. (You can also use the ExchangeService.GetUserAvailability EWS Managed API method to do this.)

    The organizer can then update their meeting with the new proposed meeting times and send the updates to all attendees by using the UpdateItem EWS operation (or the Appointment.Update EWS Managed API method).
